Documentation: Managed resources section (#1513)
crossplane/crossplane: docs/managed-resources.md

Virtual Network. External resources have specific fields for these relations, however,
they usually require the information to be supplied in different formats. In Azure
MySQL, you might be required to enter only the name of the Virtual Network while in
Azure Kubernetes, it could be required to enter a string in a specific format
that includes other information such as resource group name.

In Crossplane, users have 3 fields to refer to another resource. Here is an example
from Azure MySQL managed resource referring to a Azure Resource Group:

```yaml
spec:
  forProvider:
    resourceGroupName: foo-res-group
    resourceGroupNameRef:
      name: resourcegroup
    resourceGroupNameSelector:
      matchLabels:
        app: prod
```

In this example, the user provided only a set of labels to select a `ResourceGroup`
managed resource that already exists in the cluster via `resourceGroupNameSelector`.
Then after a specific `ResourceGroup` is selected, `resourceGroupNameRef` is filled
with the name of that `ResourceGroup` managed resource. Then in the last step,
Crossplane fills the actual `resourceGroupName` field with whatever format Azure
accepts it.

> Users are able to give the input in all three levels: selector to select via labels,

Reviewer: ```suggestion
Users are able to specify any of these three fields: 
 - selector to select via labels
 - reference to point to a determined managed resource 
 - actual value that will be submitted to the provider
```

> reference to point to a determined managed resource and actual value that will

Reviewer: ```suggestion
```

> be submitted to the provider.

Reviewer: ```suggestion
```

It's important to note that in case a reference exists, the managed resource does
not create the external resource until the referenced object is ready. In this
example, creation call of Azure MySQL Server will not be made until referenced

> `ResourceGroup` has its condition named `Ready` to be true.

Reviewer: ```suggestion
`ResourceGroup` has its `status.condition` named `Ready` to be true.
```

## Importing Existing Resources

> If you have some resources that are already provisioned in the provider, you can

Reviewer: ```suggestion
If you have some resources that are already provisioned in your cloud provider, you can
```

import them as managed resources and let Crossplane manage them. What you need to
do is to enter the name of the external resource as well as the required fields
on the managed resource. For example, let's say I have a GCP Network provisioned
from GCP console and I would like to migrate it to Crossplane. Here is the YAML
that I need to create:

```yaml
apiVersion: compute.gcp.crossplane.io/v1beta1
kind: Network
metadata:
  name: foo-network
  annotation:
    crossplane.io/external-name: existing-network
spec:
  providerRef:
    name: gcp-creds
```

Crossplane will check whether a GCP Network called `existing-network` exists, and
if it does, then the optional fields under `forProvider` will be filled with the
values that are fetched from the provider.

> Note that if a resource has required fields, you will need to fill those fields, too,

Reviewer: ```suggestion
Note that if a resource has required fields, you must fill those fields or
```

> because otherwise creation of the managed resource will be rejected. So, in those cases,

Reviewer: ```suggestion
the creation of the managed resource will be rejected. So, in those cases,
```

> you will need to enter the name of the resource as well as values of those required

Reviewer: ```suggestion
you will need to enter the name of the resource as well as all required fields as indicated in the [API Reference][api-reference] documentation.
```

fields.

## Backup and Restore

Crossplane adheres to Kubernetes conventions as much as possible and one of the
advantages we gain is backup & restore ability with tools that work with native

> Kubernetes types.

Reviewer: ```suggestion
Kubernetes types, like [Verero](https://velero.io/).
```

If you'd like to backup and restore manually, you can simply export them and save YAMLs
in your file system. When you reload them, as we've discovered in import section,

> their external nem annotation and requried fields are there and those are enough to

Reviewer: ```suggestion
their `crosssplane.io/external-name` annotation and required fields are there and those are enough to
```
